pkg/component/watch.go

```diff
@@ -38,16 +39,21 @@ type WatchParameters struct {
 	PushDiffDelay int
 }
 
-// isRegExpMatch compiles strToMatch against each of the passed regExps
-// Parameters: a string strToMatch and a list of regexp patterns to match strToMatch with
+// isGlobExpMatch compiles strToMatch against each of the passed regExps
+// Parameters: the component directory (for converting relative path expressions to absolute ones), a string strToMatch and a list of regexp patterns to match strToMatch with
 // Returns: true if there is any match else false
-func isRegExpMatch(strToMatch string, regExps []string) (bool, error) {
-	for _, regExp := range regExps {
-		matched, err := regexp.MatchString(regExp, strToMatch)
+func isGlobExpMatch(directory, strToMatch string, globExps []string) (bool, error) {
+	for _, globExp := range globExps {
+		// for glob matching with the library
+		// the relative paths in the glob expressions need to be converted to absolute paths
+		absExp := path.Join(directory, globExp)
+		pattern, err := glob.Compile(absExp)
 		if err != nil {
 			return false, err
 		}
+		matched := pattern.Match(strToMatch)
 		if matched {
+			glog.V(4).Infof("ignoring path %s because of glob rule %s", strToMatch, globExp)
```

Comment:
Can this be done in the place where we invoke this function?
Also, this function could be generic right why treat it as a function for only path matching?
Can we pass the path as a parameter in entirety to this function instead of doing the concatenation in this function?
I know at the moment this is like exclusive for push and watch so I'll leave it to you